MainEvent/main.cpp

```diff
@@ -94,6 +103,51 @@ void MoveRobot() {
   //Roboteq.sendAngles();
 }
 
+bool sortingMethod (Mat& a, Mat& b){
+    return a.Y > b.Y;
+}
+
+void matchingMethod(){
+  //so these are sorted right now
+  sort(ballCoordR.begin(), ballCoordR.end(), sortingMethod());
+  sort(ballCoordL.begin(), ballCoordL.end(), sortingMethod());
+
+  stack<Mat> correctedR;
+  stack<Mat> correctedL;
+
+  for (auto rball: ballCoordR){
+    if (rball.X < rightBound){
+      correctedR.push(rball);
+    }
+  }
+  for (auto lball: ballCoordL){
+    if (lball.X > leftBound){
+      correctedL.push(lball);
+    }
+  }
+
+  while (!correctedR.empty()){
+    Mat rball = correctedR.top();
+    while (true){
+      if (correctedL.empty()) break;
+      Mat ball = correctedL.top();
```

Comment:
```suggestion
      Mat lball = correctedL.top();
```
Keep the naming consistent